src/lib/utils/track.ts

```diff
@@ -201,3 +218,104 @@ export interface Track<P = any, S = null, T extends Schema.Global = Schema.Entit
+ * to our native `Events.postEvent` function.
+ *
+ * As an object:
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { screenTrack, Schema } from "lib/utils/track"
+ *
+ *       @screenTrack({
+ *        context_screen: Schema.PageNames.ConsignmentsWelcome,
+ *        context_screen_owner_slug: null,
+ *        context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
+ *       })
+ *
+ *       export default class Welcome extends React.Component<Props, null> {
+ *         [...]
+ *
+ * * As an function taking account of incoming props:
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { screenTrack, Schema } from "lib/utils/track"
+ *
+ *      interface Props extends ViewProperties {
+ *        [..]
+ *      }
+ *
+ *      @screenTrack<Props>(props => ({
+ *        context_screen: Schema.PageNames.ConsignmentsSubmission,
+ *        context_screen_owner_slug: props.submissionID,
+ *        context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
+ *      }))
+ *      export default class Welcome extends React.Component<Props, null> {
+ *
+ *        [...]
+ *
+ */
+export function screenTrack<P>(trackingInfo: TrackingInfo<Schema.PageView, P, null>) {
+  return _track(trackingInfo as any, {
+    dispatch: data => Events.postEvent(data),
+    dispatchOnMount: true,
+  })
+}
+
+/**
+ * ## Writing tests for your tracked code
+ *
+ * By default we mock `react-tracking`, so it's not possible to test the code easily.
+ *
+ * A good pattern for testing analytics code is to have a completely separate file
+ * for the tests. For example: `__tests__/Overview-analytics-tests.tsx`. Jest has each
+ * test file run in a unique environment, so in that file we can unmock react-tracking.
+ *
+ * Here's a full example:
+ *
+ * import { shallow } from "enzyme"
+ * import Event from "lib/NativeModules/Events"
+ * import React from "react"
+ *
+ * // Unmock react-tracking so that it will wrap our code
+ * jest.unmock("react-tracking")
+ * import Overview from "../Overview"
+ *
+ * jest.mock("lib/NativeModules/Events", () => ({ postEvent: jest.fn() }))
+ *
+ * it("calls the draft created event", () => {
+ *
+ *   // Use enzyme to render the component tree
+ *   // note that we need to `dive` into the first child component
+ *   // so that we get to the real component not the reac-tracking HOC
+ *   const overviewComponent = shallow(<Overview [...] />).dive()
+ *   const overview = overviewComponent.instance()
+ *
+ *   // Run the function which triggers the tracking call
+ *   overview.submissionDraftCreated()
+ *
+ *   // Check that the native event for the analytics call is sent
+ *   expect(Event.postEvent).toBeCalledWith({
+ *     action_name: "consignmentDraftCreated",
+ *     action_type: "success",
+ *     context_screen: "ConsignmentsOverview",
+ *     context_screen_owner_type: "ConsignmentSubmission",
+ *     owner_id: "123",
+ *     owner_slug: "123",
+ *     owner_type: "ConsignmentSubmission",
+ *   })
```

Comment:
Probably wanna reset the mock after each test to ensure that this isn’t succeeding due to another test run prior?

Reply:
Yeah, seems reasonable to me - added